src/components/interactive-map.js

```diff
@@ -78,11 +100,57 @@ export default class InteractiveMap extends PureComponent {
     };
   }
 
+  componentDidMount() {
+    // Register event handlers
+    const {eventCanvas} = this.refs;
+    const {mapControls} = this.props;
+
+    this._eventManager = new EventManager(eventCanvas);
+
+    mapControls.events.forEach(event => this._eventManager.on(event, this._handleEvent));
+  }
+
+  componentWillUnmount() {
+    if (this._eventManager) {
+      // Must destroy because hammer adds event listeners to window
+      this._eventManager.destroy();
+    }
+  }
+
+  _handleEvent(event) {
+    const {mapControls} = this.props;
```

Comment:
Indeed... though `mapControls` only extracts states that it recognizes.
The scenario I'm trying to accommodate is custom callbacks (say I need a custom map control that has callbacks `onMove`, `onZoom` etc.)